src/Integration.Vsix/SonarLintTagger/TaggerProvider.cs

```diff
@@ -112,24 +115,17 @@ internal sealed class TaggerProvider : IViewTaggerProvider, ITableDataSource, II
 
             if (detectedLanguages.Any())
             {
-                lock (taggers)
-                {
-                    if (!taggers.ExistsForFile(textDocument.FilePath))
-                    {
-                        return new IssueTagger(dte, this, buffer, textDocument, detectedLanguages)
-                            as ITagger<T>;
-                    }
-                }
+                // Multiple views could have that buffer open simultaneously, so only create one instance of the tracker.
+                var issueTracker = buffer.Properties.GetOrCreateSingletonProperty(typeof(TextBufferIssueTracker),
```

Comment:
Copied from the VS Extensibility [_Error List_ ](https://github.com/Microsoft/VSSDK-Extensibility-Samples/blob/master/ErrorList/C%23/SpellCheckerProvider.cs#L61 )sample - guaranteed way to create one tracker per file.